plugins/techdocs-react/api-report.md

```diff
@@ -0,0 +1,116 @@
+
+```ts
+import { AsyncState } from 'react-use/lib/useAsync';
+import { ComponentType } from 'react';
+import { CompoundEntityRef } from '@backstage/catalog-model';
+import { Context } from 'react';
+import { Dispatch } from 'react';
+import { Entity } from '@backstage/catalog-model';
+import { Extension } from '@backstage/core-plugin-api';
+import { default as React_2 } from 'react';
+import { SetStateAction } from 'react';
+import { VersionedValue } from '@backstage/version-bridge';
+
+// @alpha
+export function createTechDocsAddon<TComponentProps>(
+  options: TechDocsAddonOptions<TComponentProps>,
+): Extension<ComponentType<TComponentProps>>;
+
+// @alpha (undocumented)
+export const defaultTechDocsReaderPageValue: TechDocsReaderPageValue;
+
+// @alpha
+export const TECHDOCS_ADDONS_WRAPPER_KEY = 'techdocs.addons.wrapper.v1';
+
+// @alpha
+export enum TechDocsAddonLocations {
+  CONTENT = 'content',
+  HEADER = 'header',
+  PRIMARY_SIDEBAR = 'primary sidebar',
+  SECONDARY_SIDEBAR = 'secondary sidebar',
+  SUBHEADER = 'subheader',
+}
+
+// @alpha
+export type TechDocsAddonOptions<TAddonProps = {}> = {
+  name: string;
+  location: TechDocsAddonLocations;
+  component: ComponentType<TAddonProps>;
+};
+
+// @alpha
+export const TechDocsAddons: React_2.ComponentType;
+
+// @public
+export type TechDocsEntityMetadata = Entity & {
+  locationMetadata?: {
+    type: string;
+    target: string;
+  };
+};
+
+// @public
+export type TechDocsMetadata = {
+  site_name: string;
+  site_description: string;
+};
+
+// @alpha (undocumented)
+export const TechDocsReaderPageContext: Context<
+  | VersionedValue<{
+      1: TechDocsReaderPageValue;
+    }>
+  | undefined
+>;
+
+// @public
+export type TechDocsReaderPageValue = {
+  metadata: AsyncState<TechDocsMetadata>;
+  entityName: CompoundEntityRef;
+  entityMetadata: AsyncState<TechDocsEntityMetadata>;
+  shadowRoot?: ShadowRoot;
+  setShadowRoot: Dispatch<SetStateAction<ShadowRoot | undefined>>;
+  title: string;
+  setTitle: Dispatch<SetStateAction<string>>;
+  subtitle: string;
+  setSubtitle: Dispatch<SetStateAction<string>>;
+  onReady?: () => void;
+};
+
+// @alpha
+export const useShadowRoot: () => ShadowRoot | undefined;
+
+// @alpha
+export const useShadowRootElements: <
+  TReturnedElement extends HTMLElement = HTMLElement,
+>(
+  selectors: string[],
+) => TReturnedElement[];
+
+// @alpha
+export const useShadowRootSelection: (wait?: number) => Selection | null;
+
+// @alpha
+export const useTechDocsAddons: () => {
+  renderComponentByName: (name: string) => React_2.ReactElement<
+    {
+      [name: string]: unknown;
+    },
+    string | React_2.JSXElementConstructor<any>
+  > | null;
+  renderComponentsByLocation: (location: TechDocsAddonLocations) =>
+    | (React_2.ReactElement<
+        {
+          [name: string]: unknown;
+        },
+        string | React_2.JSXElementConstructor<any>
+      > | null)[]
+    | null;
```

Comment:
The inferred types here look a bit scary, perhaps worth an explicit type annotation? Best prefer `JSX.Element` over the React types as well imo, since that's more TS native stuff